src/CXXTests/testSidreStorage.cc

```diff
@@ -0,0 +1,645 @@
+                                              i + 2, i + 1, i);
+
+  // std::cout << testField[0].xx()
+  //           << testField[0].xy()
+  //           << testField[0].xz() << std::endl
+  //           << testField[0].yx()
+  //           << testField[0].yy()
+  //           << testField[0].yz() << std::endl
+  //           << testField[0].zx()
+  //           << testField[0].zy()
+  //           << testField[0].zz() << std::endl;
+
+  double* rawSidreData = myData.alloc_view("SidreTest", testField)->getData();
+
+  //myData.printDataStore();
+
+  EXPECT_EQ(testField[0].xx(), rawSidreData[0]);
+  EXPECT_EQ(testField[0].xy(), rawSidreData[1]);
+  EXPECT_EQ(testField[0].xz(), rawSidreData[2]);
+}
+
+TEST(SidreDataCollectionTestDim3ThirdRankTensor, ThirdRankTensor)
+{
+  Spheral::SidreDataCollection myData;
+  int n = 10;
+
+  Spheral::NodeList<Spheral::Dim<3>> makeNodeList("test bed", n, 0);
+  Spheral::Field<Spheral::Dim<3>, Spheral::Dim<3>::ThirdRankTensor> testField("test field", makeNodeList);
+  for (int i = 0; i < n; i++)
+    testField[i] = Spheral::Dim<3>::ThirdRankTensor(i);
+
+  //std::cout << testField[0](0, 0, 0) << std::endl;
+  
+  double* rawSidreData = myData.alloc_view("SidreTest", testField)->getData();
+
+  //myData.printDataStore();
+
+  //for (int i = 0; i < n; i++)
+    EXPECT_EQ(testField[0](0,0,0), rawSidreData[0]);
+}
+
+TEST(SidreDataCollectionTestDim3FourthRankTensor, FourthRankTensor)
+{
+  Spheral::SidreDataCollection myData;
+  int n = 10;
+
+  Spheral::NodeList<Spheral::Dim<3>> makeNodeList("test bed", n, 0);
+  Spheral::Field<Spheral::Dim<3>, Spheral::Dim<3>::FourthRankTensor> testField("test field", makeNodeList);
+  for (int i = 0; i < n; i++)
+    testField[i] = Spheral::Dim<3>::FourthRankTensor(i);
+
+  //std::cout << testField[0](0, 0, 0, 0) << std::endl;
+  
+  double* rawSidreData = myData.alloc_view("SidreTest", testField)->getData();
+
+  //myData.printDataStore();
+
+  //for (int i = 0; i < n; i++)
+    EXPECT_EQ(testField[0](0, 0, 0, 0), rawSidreData[0]);
+}
+
+TEST(SidreDataCollectionTestDim3FifthRankTensor, FifthRankTensor)
+{
+  Spheral::SidreDataCollection myData;
+  int n = 10;
+
+  Spheral::NodeList<Spheral::Dim<3>> makeNodeList("test bed", n, 0);
+  Spheral::Field<Spheral::Dim<3>, Spheral::Dim<3>::FifthRankTensor> testField("test field", makeNodeList);
+  for (int i = 0; i < n; i++)
+    testField[i] = Spheral::Dim<3>::FifthRankTensor(i);
+
+  //std::cout << testField[0](0, 0, 0, 0, 0) << std::endl;
+  
+  double* rawSidreData = myData.alloc_view("SidreTest", testField)->getData();
+
+  //myData.printDataStore();
+
+  //for (int i = 0; i < n; i++)
+    EXPECT_EQ(testField[0](0, 0, 0, 0, 0), rawSidreData[0]);
+}
+
+
+int main(int argc, char** argv)
+{
+// #ifdef AXOM_USE_MPI
+//   MPI_Init(&argc, &argv);
+// #endif
+  ::testing::InitGoogleTest(&argc, argv);
+
+  int result = RUN_ALL_TESTS();
+
+// #ifdef AXOM_USE_MPI
+//   MPI_Finalize();
+// #endif
+
+  return result;
+}
+
```

Comment:
I don't think this is needed for a gtest file. CMake and googletest will make their own executable which wraps the above tests.